File: lib/restic/file.go
package restic

// contains all the data relating to a restic File
type File struct {
	Name  string `json:"name"`
	Type  string `json:"type"`
	Path  string `json:"path"`
	UID   int    `json:"uid,omitempty"`
	Gid   int    `json:"gid,omitempty"`
	Size  int    `json:"size,omitempty"`
	Mode  int    `json:"mode,omitempty"`
	Mtime string `json:"mtime,omitempty"`
	Atime string `json:"atime,omitempty"`
	Ctime string `json:"ctime,omitempty"`
}

File: src-tauri/src/restic/file.rs
// -------------------------------------------------------------------------------------------------

/// A serializable restic file, as dumped by the restic binary via `restic ls --json``  
#[derive(serde::Serialize, serde::Deserialize, Debug, Default, Clone)]
pub struct File {
    pub name: String,
    #[serde(rename = "type")]
    pub type_: String,
    pub path: String,
    #[serde(default)]
    pub uid: i64,
    #[serde(default)]
    pub gid: i64,
    #[serde(default)]
    pub size: i64,
    #[serde(default)]
    pub mode: i64,
    #[serde(default)]
    pub mtime: String,
    #[serde(default)]
    pub atime: String,
    #[serde(default)]
    pub ctime: String,
}

